src/traces/scattergl/convert.js

```diff
@@ -118,7 +118,7 @@ proto.handlePick = function(pickResult) {
         trace: this,
         dataCoord: pickResult.dataCoord,
         traceCoord: [
-            this.pickXData[index],
+            Number(this.pickXData[index]), // non-fancy scattergl has Dates
```

Comment:
It makes me a little wary, but I can't think of a reason users would mix `Date`s and date strings (at least within one trace - definitely different traces should be checked independently). The first element may not be enough though, in case it's `null` or something, but the first valid element seems OK. Especially since this trace type is specifically built for performance, I guess we can impose some sane restrictions on the users.

I haven't really looked at the architecture here, but if you're already converting the values you can just go straight to milliseconds, ignore `Date` objects altogether. Also notice that it's not exactly ISO 8601 we use, I don't know its name but I think of it as SQL date format - ie maximally it's `2016-10-13 12:34:56.789123` but we're permissive about letting you truncate it after any part.